src/snowflake/connector/s3_util.py

```diff
@@ -71,6 +73,11 @@ def create_client(stage_info, use_accelerate_endpoint=False) -> Session.resource
             ("https://" + stage_info["endPoint"]) if stage_info["endPoint"] else None
         )
 
+        # If FIPS endpoint is in us-east-1 this will still work as we are testing for stage_info['endPoint'] which
+        # is populated only in the FIPS gov deployments.
+        if use_s3_regional_url and not stage_info["endPoint"]:
```

Comment:
On line 89, we specify the `region_name` parameter in boto3 session creation, which will be used to construct url for every client connection like what the code does here, regardless of the value of `use_s3_regional_url`. I'm curious why we need an extra parameter here?

Reply:
@sfc-gh-stan This changes the way S3 url is generated by aws sdk. 
For us-east-1 regions the endpoint is built as stage.s3.amazonaws.com and this is a global URL for customers in private link environments this global URL cannot be resolved (or the customers do not want to use it). So if use_s3_regional_url is set to true then we make modification to the endpoint so that the generated S3 url is a regional URL which has region name in it and will be resolved locally.

Reply:
I see, the local changes I have (to remove S3 SDK dependency) constructs all url with the region name if given, so I guess this parameter might be deprecated in the future?

Reply:
Yeah, in that case this will be deprecated. But currently we need this for the private client customers.

Reply:
@sfc-gh-stan If you are changing all puts regional URL then that might be a breaking change. As long as we publish the change and give notice to customers that should be fine

Reply:
Could you explain why would that be a breaking change? I will approve this change and let's schedule a separate call to discuss that!